org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/references/CompoundTypeReference.java

```diff
@@ -113,6 +113,15 @@ public boolean isArray() {
 	}
 	
 	@Override
+	public LightweightTypeReference getComponentType() {
+		for(LightweightTypeReference component: expose(components)) {
```

Comment:
since we don't expose the collection here, a simple components!=null would be a bit straight and faster.